packages/assembly/index.ts

```diff
@@ -850,8 +850,12 @@ export namespace ASON {
       // all the references have been allocated, let's get entry 0 and validate type info
       let entry0 = changetype<usize>(entryMap.get(0));
       if (isReference<T>()) {
-        assert(getObjectType(entry0) == idof<T>() || __instanceof(entry0, idof<T>()));
+        // in the case of functions, idof<T>() returns 0, and breaks everything
+        if (!isFunction<T>()) {
```

Comment:
When coding this fix, I discovered that `idof<T>()` will always return `0` for functions, which means that the rtid property of functions will always be different than the id returned by the compiler. `!isFunction<T>()` is a compile time branch that will be eliminated in the case where `T` is a function and we are deserializing it.

Reply:
Are we forgetting any assertions that need to happen if it *is* a function?

Reply:
I don't know how to validate that the incoming reference is a function. Best thing we can do is assume the end user knows what they are doing.